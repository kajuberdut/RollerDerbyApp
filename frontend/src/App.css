
.App {
  text-align: center;
  /* justify-content: center; */
  /* background-color: orange; */
  height: 100vh;
  /* margin-top: 150px; */
  background-color: transparent;
  z-index: 1;

}

body {

  background-size: contain; 
  background-repeat: no-repeat;
  background-image: url("./images/skater-jam_02.svg");
  /* background-image: url("/skater-jam.svg"); */
  height: 100vh;
  background-position: right top; 
  background-attachment: fixed;
  overflow-x: hidden;
  /* overflow-y: hidden; */
  /** note that this was working but you want overflow hidden left and right I think not up and down for lists?  */

}

.track-out {
  width: 4400px;
  height: 5400px;
  border: 5px solid black;
  border-radius: 50% 50%;
  background-color: transparent;
  position: absolute; 
  top: 130%; 
  left: -25%; 
  transform: translate(-50%, -50%);
  z-index: -1; 
}

/* .track-out {
  clip-path: circle(50% at 45% 45%); /* Example: Clip to a circle centered at 45% from the top and left */

.track-in {
  width: 2200px;
  /* height: 2000px; */
  height: 3800px;
  border: 5px solid black;
  border-radius: 50% 50%;
  background-color: transparent;
  position: absolute; 
  /* top: 240%;  */
  top: 130%; 
  /* left: 60%;  */
  left: -25%; 
  transform: translate(-50%, -50%);
  z-index: -1; 
}



 @media only screen and (max-width: 870px) {
  body {
    background-image: url("./images/group_01.svg");
    background-position: bottom;

  }
  
}

.ChatIcon {
  position: fixed;
  bottom: 30px;
  right: 20px;
}


/* body {
  background-image: url('data:images/utf8skater-3-poses_01.svg,<svg ...><g stroke="red" ... /></g></svg>');
} */

/* note that the fill color is the background color of the silhoutte not the background */
/* original page color pagecolor="%23ffffff" */
/* ornage fill color style="fill:%23000000 */
/* might be because of opacity but not sure  */
/* pagecolor doesnt seem to be changing this. you may have to try something else  style="fill:%231b78f6;fill-opacity:0.980392;*/

 /* body {
  background-color: red;

 } */