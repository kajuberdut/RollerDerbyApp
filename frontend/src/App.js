import './App.css';
import React, { useState, useEffect, Fragment } from 'react';
import { BrowserRouter } from "react-router-dom";
import NavBar from "./navBar/NavBar";
import AllRoutes from "./routes/Routes";
import Loading from "./multiUse/loading/Loading";
import UserContext from "./multiUse/UserContext";
// import { UserContext } from "./multiUse/UserContext";
import FastApi from "./Api";
import useLocalStorage from "./hooks/useLocalStorage";
// ! this will probably need to be moved 
import Messages from './chats/Messages';
// import jwt from "jsonwebtoken";
// import * as jwt_decode from 'jwt-decode';
// import jwt_decode from 'jwt-decode'
// import jwt_decode, { JwtPayload } from 'jwt-decode'
import { jwtDecode } from "jwt-decode"
import ChatIcon from "./chats/ChatIcon"
import ChatList from './chats/ChatList';
import { Link } from 'react-router-dom';
import ChatDetails from './chats/ChatDetails';

export const TOKEN_STORAGE_ID = "api-token";

function App() {

  const [user, setUser] = useState(null);
  const [token, setToken] = useLocalStorage(TOKEN_STORAGE_ID);
  const [isLoading, setIsLoading] = useState(false);
  const { default: jwt_decode } = require("jwt-decode");
  const [displayMessages, setDisplayMessages] = useState(false)
  const [displayChatList, setDisplayChatList] = useState(false)
  const [displayChats, setDisplayChats] = useState(false)
  const [chatId, setChatId] = useState()
  // const [displayErr, setDisplayErr] = useState(); 



  // useEffect(() => {
  //   // get user from local storage  
  //   const storedUser = localStorage.getItem('user');
  //   console.log("storedUser *** ", storedUser)
    
  //   // if there is user in local starage set user in state 
  //   if (storedUser) {
  //     setUser(JSON.parse(storedUser));
  //   }

  //   // if there is a token we are going to get the user 
  //   if (token) {
  //     getUser();
  //   }
  // }, [token, getUser]);

  // async function getUser() {
  //   try {
  //     const { sub } = jwtDecode(token);
  //     FastApi.token = token;
  //     const user = await FastApi.getUserById(sub);
  //     setUser(user);
  //     // useLocalStorage.setItem('user', JSON.stringify(user));
  //   } catch (err) {
  //     console.error("App loadUserInfo: problem loading", err);
  //     setUser(null);
  //   }
  // }

  // useEffect(() => {
  //   const user = JSON.parse(localStorage.getItem('user'));
  
  //   if (user) {
  //     setUser(user);
  //   }
  // }, []);


  useEffect(function loadUser() {
    console.debug("App useEffect loadUserInfo", "token=", token);
    /** If token changes then API get request for user using user_id after decoding token if value of token changes rerun. */

    async function getUser() {
        if (token) {
          try {
            let { sub } = jwtDecode(token);
 
            // storing the token in static token in FastApi 
            FastApi.token = token;
            let user = await FastApi.getUserById(sub);
            // !note that local storage is not what you imported this is a built in function!
            localStorage.setItem('user', JSON.stringify(user));
            setUser(user);

          } catch (err) {
            console.error("App loadUserInfo: problem loading", err);
            setUser(null);
          }
        }
        setIsLoading(false);
        console.log("setinfoloaded true")
      }
      getUser();
    }, [token]);

    async function signup(userData) {
      try {
      let token = await FastApi.signup(userData); 
      console.log("token:", token)
    
      return { success: true};
      } catch (err) {
        console.log("signup failed", err);
        // setDisplayErr(err)
        return {success: false, err};
      }
    }

    async function login(userData) {
      console.log("userDat!!!", userData)
      try {
      let token = await FastApi.login(userData); 
      console.log("!!!!! token from signup!!!", token)
      setToken(token);
      console.log("token /setToken:", token)
      return { success: true};
      } catch (err) {
        console.log("login failed", err);
        return {success: false, err};
      }
    }

    /** Logout - clear token and user */

    function logout() {
      setToken(null);
      setUser(null);
      setDisplayMessages(false);
      localStorage.setItem('user', null);

    }

    async function updateUser(user_id, data) {
      console.log("is this the error update")
      try {
        let user = await FastApi.updateUser(user_id, data);
        // setToken(token);
        // ! note I think you want to set the new user in storage but not 100% on this
        localStorage.setItem('user', JSON.stringify(user));
        return { success: true };
      } catch (errors) {
        console.error("update failed", errors);
        return { success: false, errors };
      }
    }

    function handleMessages(userId) {
      console.log("!!!!!!!!!!!!!!!!! handleMessages is being clicked!!!!!!!!!!!!!")
      console.log("userId in App.js", userId)
      displayMessages ? setDisplayMessages(false) : setDisplayMessages(true)
    }

    function handleChat(chatId) {
      console.log("!!!!!!!!!!!!!!!!! handleChat is being clicked!!!!!!!!!!!!!")
      console.log("chatId in App.js", chatId)
      setChatId(chatId)
      displayChats ? setDisplayChats(false) : setDisplayChats(true)
    }


    function handleChatList() {
      console.log("!!!!!!!!!!!!!!!!! handleChatList is being clicked!!!!!!!!!!!!!")
      displayChatList ? setDisplayChatList(false) : setDisplayChatList(true)
    }

     /** Display isLoading if API call is has not returned */

  if (isLoading) {
    return (
        <Loading />
    )
  }


  return (
    <>
    <div className="track-out">  </div>
    <div className="track-in">  </div>
    <div className="App">
   
      <BrowserRouter>
      <UserContext.Provider value={{user, setUser}}>
        <Fragment>
          <NavBar logout={logout}/>
          {/* <main>
            <AllRoutes handleMessages={handleMessages} signup={signup} login={login} update={updateUser} getBouts={getBouts} getMixers={getMixers} getUsers={getUsers}/>
          { user && displayMessages &&  <Messages handleMessages={handleMessages} /> }
          </main> */}
          <main>
            <AllRoutes handleMessages={handleMessages} signup={signup} login={login} update={updateUser} />
          { user && displayMessages &&  <Messages handleMessages={handleMessages} /> }
          </main>
          {/* {user && <Link to={`/chats/${user.userId}`}>
            <ChatIcon className="ChatIcon"/>
          </Link> } */}
          {user && 
            <div onClick={handleChatList}><ChatIcon className="ChatIcon"/></div>
          }
          {user && displayChatList && <ChatList handleChatList={handleChatList} handleChat={handleChat} /> }
          {user && displayChats && <ChatDetails handleChat={handleChat} chatId={chatId} /> }
        </Fragment>
        </UserContext.Provider>
      </BrowserRouter>
      </div>
      </>
    
  );
  }

export default App;
