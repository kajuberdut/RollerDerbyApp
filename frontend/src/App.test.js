import { render, screen } from '@testing-library/react';
import App from './App';

// npm test App.test.js
// must be in frontend directory 

test('renders learn react link', () => {
  render(<App />);
  const navbar = screen.getByRole('navigation'); 
  expect(navbar).toBeInTheDocument();
});


test('matches snapshot', function() {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot(); 
})

