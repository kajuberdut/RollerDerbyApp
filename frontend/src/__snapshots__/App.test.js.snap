// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="track-out"
  >
      
  </div>
  <div
    class="track-in"
  >
      
  </div>
  <div
    class="App"
  >
    <div>
      <nav
        class="NavBar-Top navbar navbar-expand-md"
      >
        <div
          class="container-fluid"
        >
          <div
            style="display: flex;"
          >
            <div
              class="star"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-star "
                data-icon="star"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="font-size: 40px; margin-right: 7px;"
                viewBox="0 0 576 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M316.9 18C311.6 7 300.4 0 288.1 0s-23.4 7-28.8 18L195 150.3 51.4 171.5c-12 1.8-22 10.2-25.7 21.7s-.7 24.2 7.9 32.7L137.8 329 113.2 474.7c-2 12 3 24.2 12.9 31.3s23 8 33.8 2.3l128.3-68.5 128.3 68.5c10.8 5.7 23.9 4.9 33.8-2.3s14.9-19.3 12.9-31.3L438.5 329 542.7 225.9c8.6-8.5 11.7-21.2 7.9-32.7s-13.7-19.9-25.7-21.7L381.2 150.3 316.9 18z"
                  fill="currentColor"
                />
              </svg>
            </div>
            <a
              class="navbar-brand"
              href="/"
            >
              Conteact
            </a>
          </div>
          <ul
            class="ml-auto navbar-nav"
          >
            <li
              class="Navbar Signup nav-item"
            >
              <a
                class=""
                href="/signup"
              >
                Signup
              </a>
            </li>
            <li
              class="Navbar Admin nav-item"
            >
              <a
                class=""
                href="/login"
              >
                Login
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <main>
      <section
        class="col-md-8 Home"
      >
        <div
          class="card"
        >
          <div
            class="card-title"
          >
            <h1>
              A Roller Derby Communication Application
            </h1>
          </div>
          <div
            class="card-body"
          >
            <p
              class="card-text"
            />
            -
            <p
              class="card-text"
            >
              <a
                class="Home-Link"
                href="/login"
              >
                <button
                  class="btn btn-secondary"
                >
                  Login
                </button>
              </a>
              <a
                class="Home-Link"
                href="/signup"
              >
                <button
                  class="btn btn-secondary"
                >
                  Signup
                </button>
              </a>
            </p>
          </div>
        </div>
      </section>
    </main>
  </div>
</DocumentFragment>
`;
